Report error location in the parsed file, not only in the lens

mvidner opened the pull request:
[bsc#1077435](https://bugzilla.suse.com/show_bug.cgi?id=1077435) but **note** that the bug is for SLE12-SP3 and this one is for master. Should I backport it? (Uh, there don't seem to be any maintenance branches for this...)

As seen in https://github.com/yast/yast-network/pull/603, when there's a problem with parsing a config file, we fail to tell the user which part of the config file is at fault.

Before:
> Client call failed with 'Augeas parsing/serializing error: Iterated lens matched less than it should at  /usr/share/augeas/lenses/dist/hosts.aug:23.12-.42:' (RuntimeError)

After:
> Client call failed with 'Augeas **parsing** error: Iterated lens matched less than it should at **/etc/hosts:34:0, lens** /usr/share/augeas/lenses/dist/hosts.aug:23.12-.42:' (RuntimeError)

jreidinger commented on spec/augeas_writer_spec.rb:
```diff
@@ -13,126 +13,126 @@
     context "for collections" do
       let(:input_text) do
-        <<EOS
+        <<TEXT
```
EOS is end of string, why it is bad delimeter?

mvidner replied:
Because a new RuboCop version thinks that you should indicate the type of the string with the delimiter. WTF.

jreidinger commented on spec/augeas_parser_spec.rb:
```diff
@@ -21,7 +21,7 @@
       example_file = "invalid syntax\n"
 
-      msg = /Augeas parsing\/serializing error/
+      msg = /Augeas parsing error/
```
maybe test here that it reports correct location? like collumn and line?

mvidner commented on spec/augeas_parser_spec.rb:
```diff
@@ -44,8 +52,8 @@
 
-      msg = /Augeas parsing\/serializing error/
-      expect { subject.serialize(example_tree) }.to raise_error(msg)
+      msg = /Augeas serializing error: .* at \/etc\/sudoers::/
```
@jreidinger This shows a weakness, the line and column do not make much sense for serializing, maybe we could point to the data in the tree??? But anyway, overall it is an improvement.

mvidner commented on lib/cfa/base_model.rb:
```diff
@@ -51,7 +51,7 @@ def save(changes_only: false)
     #   depending on the used parser it may raise an error.
     def load
-      self.data = @parser.parse(@file_handler.read(@file_path))
+      self.data = @parser.parse(@file_handler.read(@file_path), @file_path)
```
Ugh, this breaks with config_files_api_grub2, whose parser expects a single argument :-/

Either we simply change the other gem too, or we allow both APIs and here in the caller dynamically check the #parse method arity and only add the file path if supported...

mvidner commented:
@mchf @jreidinger updated: found that several classes depend on the existing API so I changed the approach to be backward compatible, and added a new test for that.

mchf commented:
code wise it looks good to me. I'm going to test it in my particular use case.
